embedded-service/src/power/policy/action/policy.rs

```diff
@@ -42,23 +47,32 @@ impl<'a, S: Kind> Policy<'a, S> {
         }
     }
 
-    async fn disconnect_internal(&self) -> Result<(), Error> {
+    /// Common disconnect function used by multiple states
+    async fn disconnect_internal_no_timeout(&self) -> Result<(), Error> {
         info!("Device {} got disconnect request", self.device.id().0);
         self.device
-            .execute_device_request(device::RequestData::Disconnect)
+            .execute_device_command(device::CommandData::Disconnect)
             .await?
             .complete_or_err()?;
         self.device.set_state(device::State::Idle).await;
         self.device.exit_recovery().await;
         Ok(())
     }
 
-    /// Connect this device as a provider
-    async fn connect_provider_internal(&self, capability: PowerCapability) -> Result<(), Error> {
+    /// Common disconnect function used by multiple states
+    async fn disconnect_internal(&self) -> Result<(), Error> {
+        match with_timeout(DEFAULT_TIMEOUT, self.disconnect_internal_no_timeout()).await {
+            Ok(r) => r,
+            Err(_) => Err(Error::Timeout),
```

Comment:
nit: `with_timeout` doesn't return a `non_exhaustive` type, so the catch-all isn't necessary here. If/when a new error variant is added to the error, this will erronesously compile when it otherwise should have failed (owing to the lack of `non_exhaustive`) and we'll therefore return a timeout erroneously. Prefer an exact pattern in this case. We *want* compilation failures when breaking changes are made so that we are prevented from shipping bugs.
```suggestion
            Err(embassy_time::TimeoutError) => Err(Error::Timeout),
```

here and elsewhere

Reply:
Done.